examples/Microsoft.Spark.CSharp.Examples/Sql/Batch/UserLogProcessing.cs

```diff
@@ -0,0 +1,186 @@
+                "SELECT truespamlogs.value FROM truespamlogs");
+
+            // Explore the columns in the data we have filtered
+            // Let's try getting the number of GET requests
+            IEnumerable<Row> rows = trueSpamSql.Collect();
+            int numGetRequests = 0;
+            foreach(Row row in rows)
+            {
+                string rowstring = row.ToString();
+                numGetRequests += ParseLog(rowstring);
+            }
+
+            Console.WriteLine("Number of GET requests: " + numGetRequests);
+
+            spark.Stop();
+        }
+
+        public static bool RegTest(string logLine, string regexType)
+        {            
+            // Example Apache log line:   
+            // 64.242.88.10 - - [07/Mar/2004:16:47:12 -0800] "GET /robots.txt HTTP/1.1" 200 68
+            // 1:IP   2:client   3:user   4:date time   5:method   
+            // 6:req   7:proto   8:respcode   9:size
+            Regex useRx = new Regex(
+                    "^(\\S+) (\\S+) (\\S+) \\[([\\w:/]+\\s[+\\-]\\d{4})\\] \"(\\S+) (\\S+) (\\S+)\" (\\d{3}) (\\d+)");
+
+            // Which regex to use based on what we're filtering for
+            if(regexType == "ipfilter")
+            {
+                useRx = new Regex("^(?=10)");
+            }
+            else if(regexType == "spamfilter")
+            {
+                useRx = new Regex("\\b(?=spam)\\b");
+            }
+
+            if(useRx.IsMatch(logLine))
+            {
+                return true;
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public static int ParseLog(string logLine)
+        {
+            Match match = Regex.Match(
+                logLine, 
+                "^(\\S+) (\\S+) (\\S+) \\[([\\w:/]+\\s[+\\-]\\d{4})\\] \"(\\S+) (\\S+) (\\S+)\" (\\d{3}) (\\d+)");
+
+            int groupCtr = 0;
+            int entryCtr = 0;
+            int numGets = 0;
+
+            string[] colNames = 
+                {"ip", 
+                "client", 
+                "user", 
+                "date", 
+                "method", 
+                "endpt", 
+                "protocol", 
+                "response", 
+                "size"};
+
+            // Print out the full log line and then divide based on column
+            foreach(Group group in match.Groups)
+            {
+                if(groupCtr == 0)
+                {
+                    Console.WriteLine("Full log entry: '{0}'", group.Value);
+                    ++groupCtr;
+                    continue;
+                }
+                Console.WriteLine("    {0}: '{1}'", colNames[entryCtr], group.Value);
+
+                // Determine if it's a GET request
+                if(colNames[entryCtr] == "method")
+                {
+                    if(group.Value == "GET")
+                    {
+                        ++numGets;
+                    }
+                }
+                ++groupCtr;
```

Comment:
What purpose does this serve ?  Seems like it'll be caught by the first if statement at the start of the loop and incremented.

Reply:
Good catch, I ran the code without it and it didn't change anything. Removed!